Config documentation (#103)
microsoft/scitt-ccf-ledger: docs/configuration.md

> # A Rough Guide to SCITT Configuration

Reviewer: ```suggestion
# SCITT Configuration Guide
```

> When SCITT-CCF nodes are first deployed, they are booted with initial [node configuration](https://microsoft.github.io/CCF/main/operations/configuration.html).

Reviewer: ```suggestion
When SCITT-CCF nodes are first deployed, they are started with an initial [node configuration](https://microsoft.github.io/CCF/main/operations/configuration.html).
```

> Members registered in the inital node configuration must then be [activated](https://microsoft.github.io/CCF/main/governance/adding_member.html#activating-a-new-member).

Reviewer: ```suggestion
Members registered in the initial node configuration must then be [activated](https://microsoft.github.io/CCF/main/governance/adding_member.html#activating-a-new-member).
```

Members can then make and vote on [proposals](https://microsoft.github.io/CCF/main/governance/proposals.html) to update SCITT service configuration.

> Once SCITT is apropriately configured members can vote to [open](https://microsoft.github.io/CCF/main/governance/open_network.html#opening-the-network) the service.

Reviewer: ```suggestion
Once SCITT is appropriately configured members can vote to [open the service](https://microsoft.github.io/CCF/main/governance/open_network.html#opening-the-network).
```

> - Note: SCITT does not require CCF style `Users` to be configured.

Reviewer: ```suggestion
- Note: SCITT does not require CCF-style "users" to be configured.
```

> # SCITT Configuration

Reviewer: Should be `##` as there typically is only a single top-level heading per document.

> SCITT config can be set via the `set_scitt_configuration` action within a governance proposal. Each item in `args.configuration` within `set_scitt_configuration` is a separate configuration option. Existing configuration options are outlined in the sections below.

Reviewer: ```suggestion
SCITT configuration can be set via the `set_scitt_configuration` action within a governance proposal. Each item in `args.configuration` within `set_scitt_configuration` is a separate configuration option. Existing configuration options are outlined in the sections below.
```

> Example config proposal:

Reviewer: ```suggestion
Example configuration proposal:
```

```
{
  "actions": [
    {
      "name": "set_scitt_configuration",
      "args": {
        "configuration": {
          "authentication": <auth-snippet-omitted-for-brevity>,

>           "service_identifier": "did:web:example.com%3A8000:scitt"

Reviewer: ```suggestion
          "service_identifier": "did:web:example.com:scitt"
```

Example `set_scitt_configuration` snippet:
```
"authentication": {
  "allow_unauthenticated": True
}
```

### JWT API Authentication
If JWT authentication is enabled then API requests must include a header containing an acceptable JWT from a trusted identity provider. For more details see the [CCF documentation on JWTs](https://microsoft.github.io/CCF/main/build_apps/auth/jwt.html).
- JWT providers can be configured via the `set_jwt_issuer` action as explained in the [CCF documentation](https://microsoft.github.io/CCF/main/build_apps/auth/jwt.html#setting-up-a-token-issuer-with-manual-key-refresh).

Extra `required_claims` can be configured which must then be present in an API request's JWT for authentication to succeed.

To enable JWT authentication in SCITT, add the following config to a `set_scitt_configuration` action:
```
"authentication": {
  "allow_unauthenticated": False,
  "jwt": {
    "required_claims": {
      "foo": "bar",
        ...
    }
  }
}
```

## Service ID
The long-term stable identifier of this service, as a DID.
If set, it will be used to populate the issuer field of receipts.

Example `set_scitt_configuration` snippet:
```

> "service_identifier": "did:web:example.com%3A8000:scitt"

Reviewer: ```suggestion
"service_identifier": "did:web:example.com:scitt"
```

```

## Accepted algorithms

> List of accepted COSE signature algorithms when verifying signatures in submitted claims.

Reviewer: Should say what happens by default when this is not configured explicitly.

- Note: Items in the accepted algorithms list are case sensitive.

Example `set_scitt_configuration` snippet:
```

> "accepted_algorithms": [ES256, ES384, ES512, PS256, PS384, PS512, EDDSA]

Reviewer: ```suggestion
"accepted_algorithms": ["ES256", "ES384", "ES512", "PS256", "PS384", "PS512", "EDDSA"]
```

```

> ## Truststores

Reviewer: ```suggestion
## Trust stores
```

> SCITT has two truststores that can be configured; `x509_roots`, and `did_web_tls_roots`.

Reviewer: ```suggestion
SCITT has two trust stores that can be configured: `x509_roots` and `did_web_tls_roots`.
```
